components/script/dom/html/htmldetailselement.rs

```diff
@@ -253,6 +257,8 @@ impl VirtualMethods for HTMLDetailsElement {
 
             let counter = self.toggle_counter.get() + 1;
             self.toggle_counter.set(counter);
+            let old_state = if self.Open() { "closed" } else { "open" };
+            let new_state = if self.Open() { "open" } else { "closed" };
```

Comment:
Nit to only call `self.Open()` once:

```suggestion
            let (old_state, new_state) = if self.Open() { ("closed", "open" ) } else { ("open", "closed") };
```

Reply:
It doesn't do that simply because I decided not to do that in this PR. It definitely should as a follow up.